src/amber/support/predicates.cr

```diff
@@ -0,0 +1,236 @@
+module Amber
+  module Support
+    module Predicates
+      module String
+        macro included
+                    def str?
```

Comment:
you'll have to do the indentation yourself here...

but fist, why put this in a macro hook instead of puting this in the module String directly?